cirq/sim/sampler.py

```diff
@@ -38,6 +38,21 @@ def run(
         Returns:
             TrialResult for a run.
         """
+        circuit = (program if isinstance(program, circuits.Circuit) else
+                   program.to_circuit())
+
+        if protocols.is_parameterized(circuit):
+            unresolved_params = protocols.check_parameters(
+                circuit, study.ParamResolver(param_resolver))
+            if unresolved_params is not None:
```

Comment:
Why the second condition? The execution is still going to fail if this condition isn't met.

Reply:
Hi @Strilanc - thanks very much for this question and for your time reviewing the code.

The reason that I added that second condition is certain tests were failing without it. For example, the following code did pass an unresolved parameter to `run`, though also passed a set of valid parameters along with it that were eventually captured by `run_sweep`:

    def test_run_param_resolver(dtype):
        q0, q1 = cirq.LineQubit.range(2)
        simulator = cirq.DensityMatrixSimulator(dtype=dtype)
        for b0 in [0, 1]:
            for b1 in [0, 1]:
                circuit = cirq.Circuit.from_ops((cirq.X**sympy.Symbol('b0'))(q0),
                                                (cirq.X**sympy.Symbol('b1'))(q1),
                                                cirq.measure(q0),
                                                cirq.measure(q1))
                param_resolver = cirq.ParamResolver({'b0': b0, 'b1': b1})
                result = simulator.run(circuit, param_resolver=param_resolver)

I would certainly defer to folks who are closer to this than me, but I wasn't sure if we want code like the above to work when using `run` or if it would be best to have this fail when using `run` and to address this by revising the tests to use `run_sweep`.

Any thoughts on this would be very much appreciated.